astropy/coordinates/stokes_coord.py

```diff
@@ -0,0 +1,245 @@
+
+
+class StokesCoord(ShapedLikeNDArray):
+    """
+    A representation of stokes coordinates with helpers for converting to profile names.
+
+    Parameters
+    ----------
+    stokes_values : array-like
+        The numeric values representing stokes coordinates.
+    """
+
+    info = StokesCoordInfo()
+
+    def __init__(self, stokes, copy=False):
+        data = np.asanyarray(stokes)
+        if data.dtype.kind == "U":
+            self._data = self._from_symbols(data)
+        else:
+            self._data = data.copy() if copy and data is stokes else data
+
+    @property
+    def shape(self):
+        return self._data.shape
+
+    @property
+    def value(self):
+        return self._data
+
+    @property
+    def quantity(self):
+        return Quantity(self._data)
+
+    def _apply(self, method, *args, **kwargs):
+        cls = type(self)
+
+        if callable(method):
+            new = cls(method(self._data, *args, **kwargs))
+        else:
+            new = cls(getattr(self._data, method)(*args, **kwargs))
+
+        # Copy other 'info' attr only if it has actually been defined.
+        # See PR #3898 for further explanation and justification, along
+        # with Quantity.__array_finalize__
+        if "info" in self.__dict__:
+            new.info = self.info
+
+        return new
+
+    @staticmethod
+    def _from_symbols(symbols):
+        """
+        Construct a StokesCoord from strings representing the stokes symbols.
+        """
+        values_array = np.full_like(
+            symbols, UNKNOWN_STOKES_VALUE, dtype=int, subok=False
+        )
+        for stokes_value, symbol in STOKES_VALUE_SYMBOL_MAP.items():
+            values_array[symbols == symbol.symbol] = stokes_value
+
+        if (unknown_values := np.equal(values_array, UNKNOWN_STOKES_VALUE)).any():
+            raise ValueError(
+                f"Unknown stokes symbols present in the input array: {np.unique(symbols[unknown_values])}"
+            )
+
+        return values_array
+
+    @property
+    def _stokes_values(self):
+        """
+        A representation of the coordinate as integers.
+        """
+        # Note we unbroadcast and re-broadcast here to prevent the new array
+        # using more memory than the old one.
+        return np.broadcast_to(np.round(unbroadcast(self._data)), self.shape)
+
+    @property
+    def symbol(self):
+        """
+        The coordinate represented as strings
+        """
+        known_symbols = tuple(
+            ["?"] + [s.symbol for s in STOKES_VALUE_SYMBOL_MAP.values()]
+        )
+        max_len = np.max([len(s) for s in known_symbols])
+
+        # Note we unbroadcast and re-broadcast here to prevent the new array
+        # using more memory than the old one.
+        symbolarr = np.full(unbroadcast(self.value).shape, "?", dtype=f"<U{max_len}")
+
+        for value, symbol in STOKES_VALUE_SYMBOL_MAP.items():
+            symbolarr[np.round(unbroadcast(self.value)) == value] = symbol.symbol
```

Comment:
You could pull out
`unbroadcasted_values = np.round(unbroadcast(self.value))`
and then use it here and in the definition of `symbolarr`